### PR TITLE
Interactivity API: Added comments for directive parsing

### LINKED ISSUE #61455: Interactivity API: Improve readability in directives check function inside `vdom.ts`
This section could probably use some comments about how strings are being split up and what we expect to find in each part. We've even got a constant `wp` that's the `directivePrefix` (that also appears in the middle of the data attribute).

Honestly, some kind of description of what the parts of a directive are and a refactor to rename things accordingly would be helpful. `prefix` and `suffix` don't really make sense:

```

nothing?       suffix is everything after `prefix--`… I guess this makes sense
   |              |
vvvvvvv       vvvvvvvvvv   
data-wp-bind--on--change
        ^^^^
          |
      "prefix" in the middle?
```

Maybe we could talk about these things like the "directive name" or "directive type", then the rest of it could be some kind of input to the directive? I think we use these prefix/suffix names across the package and it seems confusing.

---

The bug fix here where we try to access a `null` match is good 👍

_Originally posted by @sirreal in https://github.com/WordPress/gutenberg/pull/61249#discussion_r1592540765_

### PR DESCRIPTION
**Pull request by amitraj2203**
## What?
Fixes: #61455 

## Why?
It adds comments for directive parsing

### CODE REVIEW

**Review by cbravobernal — CHANGES_REQUESTED** [review]
Hi there!

I was thinking about doing a refactor (setting the variables names with what they should be, instead of 'prefix' or 'suffix' rather than adding code comments 😅 

Thanks for contributing on this issue!

**Comment by amitraj2203** [comment]
@cbravobernal I've made the changes you suggested! Now, instead of using 'prefix' and 'suffix', I've renamed the variables more clearly.

Thanks!

File: packages/interactivity/src/vdom.ts [hunk 1]
```diff
@@ -116,9 +116,13 @@ export function toVdom( root ) {
 		}
 
 		if ( directives.length ) {
+			// Reduce the directives array to build the __directives object.
 			props.__directives = directives.reduce(
-				( obj, [ name, ns, value ] ) => {
-					const directiveMatch = directiveParser.exec( name );
+				// The reducer function accumulates the __directives object.
+				( obj, [ directiveName, namespace, inputValue ] ) => {
+					// Check if the directive name matches the expected format.
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]:
```suggestion
```
Same here.

File: packages/interactivity/src/vdom.ts [hunk 2]
```diff
@@ -116,9 +116,13 @@ export function toVdom( root ) {
 		}
 
 		if ( directives.length ) {
+			// Reduce the directives array to build the __directives object.
 			props.__directives = directives.reduce(
-				( obj, [ name, ns, value ] ) => {
-					const directiveMatch = directiveParser.exec( name );
+				// The reducer function accumulates the __directives object.
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]:
```suggestion
```
Same here.

File: packages/interactivity/src/vdom.ts [hunk 3]
```diff
@@ -127,18 +131,22 @@ export function toVdom( root ) {
 							SCRIPT_DEBUG === true
 						) {
 							// eslint-disable-next-line no-console
-							console.warn( `Invalid directive: ${ name }.` );
+							console.warn(
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]: Let's wait for https://github.com/WordPress/gutenberg/pull/61409 to be merged and update with its function.

File: packages/interactivity/src/vdom.ts [hunk 4]
```diff
@@ -127,18 +131,22 @@ export function toVdom( root ) {
...
-					obj[ prefix ] = obj[ prefix ] || [];
-					obj[ prefix ].push( {
-						namespace: ns ?? currentNamespace(),
-						value,
-						suffix,
+					// Splitting the directive name into directive type and input.
+					const directiveType = directiveMatch[ 1 ] || '';
+					const directiveInput = directiveMatch[ 2 ] || 'default';
+
+					// Creating or updating the array for the specific directive type in the directives object.
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]:
```suggestion
```
Same here

File: packages/interactivity/src/vdom.ts [hunk 5]
```diff
@@ -127,18 +131,22 @@ export function toVdom( root ) {
...
 					}
-					const prefix = directiveMatch[ 1 ] || '';
-					const suffix = directiveMatch[ 2 ] || 'default';
-
-					obj[ prefix ] = obj[ prefix ] || [];
-					obj[ prefix ].push( {
-						namespace: ns ?? currentNamespace(),
-						value,
-						suffix,
+					// Splitting the directive name into directive type and input.
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]:
```suggestion
```
Same here

File: packages/interactivity/src/vdom.ts [hunk 6]
```diff
@@ -127,18 +131,22 @@ export function toVdom( root ) {
...
+					const directiveType = directiveMatch[ 1 ] || '';
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]:
```suggestion
					const directive = directiveMatch[ 1 ] || '';
```
I think directive is a better name.

File: packages/interactivity/src/vdom.ts [hunk 7]
```diff
@@ -127,18 +131,22 @@ export function toVdom( root ) {
...
+					const directiveInput = directiveMatch[ 2 ] || 'default';
```
Comment (cbravobernal) [inline packages/interactivity/src/vdom.ts]: Could be "event" or "directiveEvent" a better name for this?

Not sure, cause they are events in `data-wp-on-document` or `data-wp-on-window` or `data-wp-on`. But can also be styles or classes in `data-wp-class--` or `data-wp--style`. Even attributes for `data-wp-bind`

What do you think about it? Let's ping some folks with experience developing with the Interactivity API to get some feedback.

Ping: @ryanwelcher , @luisherranz , @DAreRodz or @sethrubenstein

**Review by cbravobernal — CHANGES_REQUESTED** [review]
I'm leaving this on `Request changes` to wait for https://github.com/WordPress/gutenberg/pull/61409 to be merged and also to get some more feedback about the naming of those vars.

**Comment by amitraj2203** [comment]
Hi @cbravobernal I have made changes as per your suggestion. Now waiting for replies from other folks.

**Comment by cbravobernal** [comment]
Sorry @amitraj2203 but we are not going to merge this PR. The prefix word is being used through all the Interactivity codebase, and is a bigger refactor than expected.

**Pull request closed without merging**